pkg/apis/jvmbuildservice/v1alpha1/jbsconfig_types.go

```diff
@@ -114,6 +115,7 @@ type ImageRegistry struct {
 type MavenDeployment struct {
 	Username   string `json:"username,omitempty"`
 	Repository string `json:"repository,omitempty"`
+	OnlyDeploy bool   `json:"onlyDeploy,omitempty"` // we don't want cache attempting to download from empty repo
```

Comment:
What happens without this?

Reply:
Without this the maven env variables get picked up by the cache pod too and it'll attempt to download every dependency for maven builds from our empty reposilite repo. If we _really_ want to we could potentially configure reposilite to mirror https://repo.maven.apache.org/maven2, but that would require creating config on a local machine and then mounting it as a volume and probably isn't worth it for mere testing. But do let me know if you think otherwise. :slightly_smiling_face:

Reply:
First of all, massive work, thanks @tecarter94 ! Secondly, I think we will remove the cache soonish anyway, so the flag `OnlyDeply` will be removed soon as well? I think we can leave it as it is and make updates in subsequent PRs. WDYT @rnc ? Thanks!

Reply:
I agree with having `onlyDeploy` in for now and revisting to how JBS uses the cache later.

Reply:
Makes sense. Happy to leave the flag in place for now. :+1: